src/mixing_fab.h

```diff
@@ -0,0 +1,151 @@
+
+#ifndef CYCAMORE_SRC_MIXING_FAB_H_
+#define CYCAMORE_SRC_MIXING_FAB_H_
+
+#include <string>
+#include "cyclus.h"
+#include "cycamore_version.h"
+
+namespace cycamore {
+  
+  /// MixingFab basically mixes N streams with fixed, static, user-specified
+  /// ratios into a single output stream. The MixingFab has N input inventories:
+  /// one for each streams to be mixed, and one output stream. The supplying of
+  /// mixed material is constrained by available inventory of mixed material
+  /// quantities.
+class MixingFab : public cyclus::Facility {
+#pragma cyclus note {   	  \
+    "niche": "mixing facility",				  \
+    "doc":								\
+    "MixingFab basically mixes N streams with fixed, static, user-specified" \
+    " ratios into a single output stream. The MixingFab has N input inventories:"\
+    " one for each streams to be mixed, and one output stream. The supplying of"\
+    " mixed material is constrained by available inventory of mixed material"\
+    " quantities.", \
+    }
+  
+  friend class MixingFabTest;
+
+  public:
+  MixingFab(cyclus::Context* ctx);
+  virtual ~MixingFab() {};
+  
+  virtual void Tick();
+  virtual void Tock(){};
+  virtual void EnterNotify();
+  
+  virtual void AcceptMatlTrades(const std::vector<std::pair<
+                                cyclus::Trade<cyclus::Material>, cyclus::Material::Ptr> >& responses);
+  
+  virtual std::set<cyclus::RequestPortfolio<cyclus::Material>::Ptr>
+      GetMatlRequests();
+
+
+  #pragma cyclus clone
+  #pragma cyclus initfromcopy
+  #pragma cyclus infiletodb
+  #pragma cyclus initfromdb
+  #pragma cyclus schema
+  #pragma cyclus annotations
+  #pragma cyclus snapshot
+  // the following pragmas are ommitted and the functions are written
+  // manually in order to handle the vector of resource buffers:
+  //
+  //     #pragma cyclus snapshotinv
+  //     #pragma cyclus initinv
+  
+  virtual cyclus::Inventories SnapshotInv();
+  virtual void InitInv(cyclus::Inventories& inv);
+
+  
+  protected:
+    
+    
+  #pragma cyclus var { \
+    "doc": "Ordered list of commodities on which to request material stream" \
+    " material, each commodity corresponds to an input stream.", \
+    "uilabel": "Stream Commodities", \
+    "uitype": ["oneormore", "incommodity"], \
+  }
+  std::vector<std::string> in_commods;
```

Comment:
Does this variable have the underscore for other archtypes? I thought it was just `incommods` elsewhere...